pkg/helpers/identifier_string_parser.go

```diff
@@ -25,8 +24,126 @@ func parseIdentifierStringWithOpts(identifier string, opts func(*csv.Reader)) ([
 	if err != nil {
 		return nil, fmt.Errorf("unable to read identifier: %s, err = %w", identifier, err)
 	}
+	if lines == nil {
+		return make([]string, 0), nil
+	}
 	if len(lines) != 1 {
 		return nil, fmt.Errorf("incompatible identifier: %s", identifier)
 	}
 	return lines[0], nil
 }
+
+func ParseIdentifierString(identifier string) ([]string, error) {
+	return parseIdentifierStringWithOpts(identifier, func(r *csv.Reader) {
+		r.Comma = IdDelimiter
+	})
+}
+
+func ParseResourceIdentifier(identifier string) []string {
+	if identifier == "" {
+		return make([]string, 0)
+	}
+	return strings.Split(identifier, string(ResourceIdDelimiter))
+}
+
+func EncodeResourceIdentifier(parts ...string) string {
+	return strings.Join(parts, string(ResourceIdDelimiter))
+}
+
+func parseAccountObjectIdentifier(identifier string, parser identifierParsingFunc) (sdk.AccountObjectIdentifier, error) {
+	parts, err := parser(identifier)
+	if err != nil {
+		return sdk.AccountObjectIdentifier{}, err
+	}
+	if len(parts) != 1 {
+		return sdk.AccountObjectIdentifier{}, fmt.Errorf(`unexpected number of parts %d in identifier %s, expected 1 in a form of "<account_object_name>"`, len(parts), identifier)
+	}
+	return sdk.NewAccountObjectIdentifier(parts[0]), nil
+}
+
+func ParseAccountObjectIdentifier(identifier string) (sdk.AccountObjectIdentifier, error) {
+	return parseAccountObjectIdentifier(identifier, ParseIdentifierString)
+}
+
+func parseDatabaseObjectIdentifier(identifier string, parser identifierParsingFunc, delimiter rune) (sdk.DatabaseObjectIdentifier, error) {
+	parts, err := parser(identifier)
+	if err != nil {
+		return sdk.DatabaseObjectIdentifier{}, err
+	}
+	if len(parts) != 2 {
+		return sdk.DatabaseObjectIdentifier{}, fmt.Errorf(`unexpected number of parts %d in identifier %s, expected 2 in a form of "<database_name>%c<database_object_name>"`, len(parts), identifier, delimiter)
+	}
+	return sdk.NewDatabaseObjectIdentifier(parts[0], parts[1]), nil
+}
+
+func ParseDatabaseObjectIdentifier(identifier string) (sdk.DatabaseObjectIdentifier, error) {
+	return parseDatabaseObjectIdentifier(identifier, ParseIdentifierString, IdDelimiter)
+}
+
+func parseSchemaObjectIdentifier(identifier string, parser identifierParsingFunc, delimiter rune) (sdk.SchemaObjectIdentifier, error) {
+	parts, err := parser(identifier)
+	if err != nil {
+		return sdk.SchemaObjectIdentifier{}, err
+	}
+	if len(parts) != 3 {
+		return sdk.SchemaObjectIdentifier{}, fmt.Errorf(`unexpected number of parts %[1]d in identifier %[2]s, expected 3 in a form of "<database_name>%[3]c<schema_name>%[3]c<schema_object_name>"`, len(parts), identifier, delimiter)
+	}
+	return sdk.NewSchemaObjectIdentifier(parts[0], parts[1], parts[2]), nil
+}
+
+func ParseSchemaObjectIdentifier(identifier string) (sdk.SchemaObjectIdentifier, error) {
```

Comment:
To better grasp the impact of the change I would like to see the example usage (in at least one resource; any of the ParseX functions, or example for all of them). Also, the encode one.

It will help to understand what we are replacing and how it looks after the transition

Reply:
Used encode/parse of the resource identifier in a few places, and used mostly parser for external object identifier in shared+secondary databases (also revealing possible root causes of their flakiness 😞).